index.js

```diff
@@ -4,15 +4,103 @@
 
   /*istanbul ignore next*/
   if(module && typeof module.exports !== 'undefined'){
-    module.exports = f();
+    module.exports = f(require('sanctuary-type-classes'), require('sanctuary-type-identifiers'));
   }else{
-    global.concurrify = f();
+    global.concurrify = f(global.sanctuaryTypeClasses, global.sanctuaryTypeIdentifiers);
   }
 
-}(/*istanbul ignore next*/(global || window || this), function(){
+}(/*istanbul ignore next*/(global || window || this), function(Z, type){
 
   'use strict';
 
-  return function(){};
+  var $alt = 'fantasy-land/alt';
+  var $ap = 'fantasy-land/ap';
+  var $map = 'fantasy-land/map';
+  var $of = 'fantasy-land/of';
+  var $zero = 'fantasy-land/zero';
+  var $$type = '@@type';
+
+  var ordinal = ['first', 'second', 'third', 'fourth', 'fifth'];
+  var show = Z.toString;
+
+  function invalidArgument(it, at, expected, actual){
+    throw new TypeError(
+      it
+      + ' expects its '
+      + ordinal[at]
+      + ' argument to '
+      + expected
+      + '\n  Actual: '
+      + show(actual)
+    );
+  }
+
+  function invalidContext(it, actual){
+    throw new TypeError(
+      it + ' was invoked outside the context of a Concurrently. \n  Called on: ' + show(actual)
+    );
+  }
+
+  function getReprType(Repr){
+    return Repr[$$type] || 'concurrify/' + (Repr.name || 'Anonymous');
+  }
+
+  function computeType(type){
+    var parts = type.split('/');
+    return parts[0] + '/Concurrent' + parts[1];
```

Comment:
> the type representative's @@type property (the type identifier) must be a string primitive, **ideally '<npm-package-name>/<type-name>'**.

i.e. `type` might not contain `/`,  I think safer way would be to just concat 'Concurrent' in the end